### ISSUE TITLE
站位是否影响战列开炮顺序问题

### ISSUE THREAD

**Issue by joyxsr** [AzurLaneTools/AzurLaneCDTool#10]
尽管在这一图片中，前卫号战列舰的实际CD为18.719，比武藏号战列舰的实际CD 18.793要快，但计算器仍然认为武藏号的开炮时间比前卫更早。这一想法的依据可能是武藏是旗舰，而前卫是僚舰，当两者均装填完成时，武藏先开炮。然而实践并不支持这一观点。在meta作战、月度boss战中，“谁先开炮”完全由“谁先装填好”决定，而不受站位影响。
![Snipaste_2022-10-24_15-03-45](https://user-images.githubusercontent.com/64903778/197467301-e9c04e8e-8411-4ba1-b079-6fb7fe4a46a8.png)

**Comment by AutumnSun1996** [comment]
我开发时是根据wiki里的说明来的, 也就是站位会影响开炮顺序, 我记得是有测试支持这一理论的.

给出的这个案例中, 实际开炮顺序和结果不符, 可能是其他内容的影响吗?
(比如公共CD计算逻辑, 已确认存在一些问题但暂时无从修改)

目前应该可以通过手动调整顺序避免这个问题. 另外, 有更多的测试来验证具体规则吗?

**Comment by Resonance1983** [comment]
[我也遇到了相同的问题，但是在这次活动的boss模拟战中表明仍然是按照实际cd计算开炮](url)

![2dfc1e3a841755b927477dcde89fddd](https://github.com/AzurLaneTools/az-cd/assets/58250140/f3ff4fda-fb68-41f1-98b4-29f4bf48d205)
![61766ae745b9bcba419f6de5271f2ab](https://github.com/AzurLaneTools/az-cd/assets/58250140/7371dd6a-887f-4f59-ac9a-5c0eb44fa223)

**Comment by Hanare** [comment]
是的 我测试出来的结果也是这样 只和谁先装填好有关 与位置无关 测试了六种不同站位与新老ui（没有付费ui所以没有测试） 结果均无区别

**Comment by Hanare** [comment]
哦 我大概摸清楚了 如果是类似预装填这种没有任何cd差的多个战列同时装填好 此时按照旗舰>上僚舰>下僚舰顺序开火
如果是因为公共cd导致的在还没有下一轮开火前剩下来的两个人都装填好了 此时开火顺序仅与谁先装填好有关 与位置无关
实战中前者场景罕见（仅三个预装填时 剩下来的场景就是各种因素导致炮cd一模一样 很罕见）后者比较常见（如测试场景中的武藏快于狮快于2234 武藏先开炮 此时狮+2234剩余cd<公共cd1s 则狮先装填好狮先开炮 2234后装填好后开炮 与摆放位置无关）
而你所开发中写的第二种情况还是按照位置而不是cd顺序计算 稍作修改即可，即仅当多个预装填或多个战列cd一模一样时按照位置开火 因公共cd积压的炮则按cd顺序开火
